com.vrcfury.vrcfury/Editor/VF/Service/OnDemandSyncService.cs

```diff
@@ -0,0 +1,111 @@
+        [VFAutowired] private readonly MathService math;
+        [VFAutowired] private readonly DirectBlendTreeService directTree;
+        [VFAutowired] private readonly SmoothingService smoothingService;
+        
+        private readonly List<VFAFloat> onDemandSyncParameters = new List<VFAFloat>();
+        
+        public void SetParameterOnDemandSync(VFAFloat parameter) {
+            onDemandSyncParameters.Add(parameter);
+        }
+
+        [FeatureBuilderAction(FeatureOrder.OnDemandSync)]
+        public void Apply() {
+            if (onDemandSyncParameters.Count == 0) return;
+            
+            var fx = manager.GetController(VRCAvatarDescriptor.AnimLayerType.FX);
+            var layers = fx.GetRaw().GetLayers();
+            foreach (var param in onDemandSyncParameters) {
+                manager.GetParams().GetParam(param.Name()).networkSynced = false;
+            }
+            var pointer = fx.NewInt("SyncPointer", synced: true);
+            var localPointer = fx.NewInt("LocalPointer"); // Use a local only pointer to avoid race condition
+            var data = fx.NewFloat("SyncData", synced: true);
+            var sending = fx.NewFloat("IsSending");
+
+            var txLayer = fx.NewLayer("Send");
+            var txIdle = txLayer.NewState("Idle");
+            txIdle.Drives(sending, 0, true);
+            var txAdder = txLayer.NewState("Adder");
+            var txResetter = txLayer.NewState("Reset");
+            txResetter.Drives(localPointer, 0, local: true)
+                .TransitionsToExit()
+                .When(fx.Always());
+            txAdder.TransitionsTo(txResetter)
+                .When(localPointer.IsGreaterThan(onDemandSyncParameters.Count - 1)); // always > 0
+            txAdder.DrivesDelta(localPointer, 1)
+                .TransitionsToExit()
+                .When(fx.Always());
+            
+            // instant sync on change detection
+            for (int i = 0; i < onDemandSyncParameters.Count; i++) {
+                var src = onDemandSyncParameters[i];
+                var lastSrc = fx.NewFloat($"last{src.Name()}");
+                directTree.Add(math.MakeCopier(src, lastSrc)); 
+                var srcDiff = math.Subtract(src, lastSrc);
+                var maintain = math.MakeMaintainer(srcDiff); // Maintain while in send animation
+                directTree.Add(sending, maintain);
+                
+                var sendInstantState = txLayer.NewState($"Send Instant {onDemandSyncParameters[i]}");
+                sendInstantState
+                    .DrivesCopy(data, src)
+                    .Drives(pointer, i)
+                    .Drives(sending, 1, true)
+                    .TransitionsToExit()
+                    .WithTransitionExitTime(0.1f)
+                    .When(fx.Always());
+                txIdle.TransitionsTo(sendInstantState)
+                    .When(fx.IsLocal().IsTrue().And(srcDiff.IsGreaterThan(0).Or(srcDiff.IsLessThan(0))));
```

Comment:
I have some concern that this will deadlock on the first variable if it's being animated or constantly moved or otherwise. Not sure there's a great solution to that though.

Reply:
I think the two sources for that are OSC and Parameter Drivers. We can skip parameters that get driven by a Parameter Driver